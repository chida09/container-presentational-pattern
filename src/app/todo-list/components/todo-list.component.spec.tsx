import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TodoListComponent } from './todo-list.component';

const mockTodo = [
  {
    id: 1,
    title: 'Task 1',
    description: 'Description 1',
    dueDate: '2022-01-01',
    isCompleted: false,
  },
  {
    id: 2,
    title: 'Task 2',
    description: 'Description 2',
    dueDate: '2022-01-02',
    isCompleted: true,
  },
];

describe('TodoListComponent', () => {
  it('should render', () => {
    const component = render(<TodoListComponent todo={mockTodo} />)
    expect(component).toMatchSnapshot()
  })

  it('should form is submitted when "タスクを追加" button click', async () => {
    const user = userEvent.setup();
    const onSaveMock = jest.fn();
    render(<TodoListComponent todo={mockTodo} onSave={onSaveMock} />);

    const titleInput = screen.getByLabelText(/title/i);
    const descriptionInput = screen.getByLabelText(/description/i);
    const dueDateInput = screen.getByLabelText(/dueDate/i);
    const addTaskButton = screen.getByRole('button', { name: 'タスクを追加' });

    await user.type(titleInput, '買い物に行く');
    await user.type(descriptionInput, '牛乳、卵を購入する');
    await user.type(dueDateInput, '2023-04-30');

    await user.click(addTaskButton);

    await waitFor(() => {
      expect(onSaveMock).toHaveBeenCalledTimes(1);
      expect(onSaveMock).toHaveBeenCalledWith({
        title: '買い物に行く',
        description: '牛乳、卵を購入する',
        dueDate: '2023-04-30',
      });
    });
  });

  // TODO: バリデーションのテストを追加する
});
