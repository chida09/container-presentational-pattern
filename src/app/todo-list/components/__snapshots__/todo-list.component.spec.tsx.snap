// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListComponent should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-gtqm8w"
      >
        <section>
          <form>
            <h2
              class="css-aqmdkm"
            >
              新規作成
            </h2>
            <table
              class="css-tv8nwf"
            >
              <thead
                class="css-14fr7p1"
              >
                <tr>
                  <th
                    class="css-5r049v"
                  >
                    <label
                      for="title"
                    >
                      title
                    </label>
                  </th>
                  <th
                    class="css-5r049v"
                  >
                    <label
                      for="description"
                    >
                      description
                    </label>
                  </th>
                  <th
                    class="css-5r049v"
                  >
                    <label
                      for="dueDate"
                    >
                      dueDate
                    </label>
                  </th>
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td
                    class="css-l9u794"
                  >
                    <input
                      class="css-1lxapgb"
                      id="title"
                      name="title"
                      placeholder="買い物に行く"
                      type="text"
                    />
                  </td>
                  <td
                    class="css-l9u794"
                  >
                    <input
                      class="css-1lxapgb"
                      id="description"
                      name="description"
                      placeholder="牛乳、卵を購入する"
                      type="text"
                    />
                  </td>
                  <td
                    class="css-l9u794"
                  >
                    <input
                      class="css-1lxapgb"
                      id="dueDate"
                      name="dueDate"
                      type="date"
                    />
                  </td>
                </tr>
              </tbody>
            </table>
            <div
              class="css-8c9rmk"
            >
              <button
                class="css-e26wfq"
                type="submit"
              >
                タスクを追加
              </button>
            </div>
          </form>
        </section>
        <section>
          <h2
            class="css-aqmdkm"
          >
            一覧
          </h2>
          <table
            class="css-tv8nwf"
          >
            <thead
              class="css-14fr7p1"
            >
              <tr>
                <th
                  class="css-5r049v"
                >
                  id
                </th>
                <th
                  class="css-5r049v"
                >
                  title
                </th>
                <th
                  class="css-5r049v"
                >
                  description
                </th>
                <th
                  class="css-5r049v"
                >
                  dueDate
                </th>
                <th
                  class="css-5r049v"
                >
                  isCompleted
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="css-l9u794"
                >
                  1
                </td>
                <td
                  class="css-l9u794"
                >
                  Task 1
                </td>
                <td
                  class="css-l9u794"
                >
                  Description 1
                </td>
                <td
                  class="css-l9u794"
                >
                  2022-01-01
                </td>
                <td
                  class="css-l9u794"
                >
                  false
                </td>
              </tr>
              <tr>
                <td
                  class="css-l9u794"
                >
                  2
                </td>
                <td
                  class="css-l9u794"
                >
                  Task 2
                </td>
                <td
                  class="css-l9u794"
                >
                  Description 2
                </td>
                <td
                  class="css-l9u794"
                >
                  2022-01-02
                </td>
                <td
                  class="css-l9u794"
                >
                  true
                </td>
              </tr>
            </tbody>
          </table>
        </section>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-gtqm8w"
    >
      <section>
        <form>
          <h2
            class="css-aqmdkm"
          >
            新規作成
          </h2>
          <table
            class="css-tv8nwf"
          >
            <thead
              class="css-14fr7p1"
            >
              <tr>
                <th
                  class="css-5r049v"
                >
                  <label
                    for="title"
                  >
                    title
                  </label>
                </th>
                <th
                  class="css-5r049v"
                >
                  <label
                    for="description"
                  >
                    description
                  </label>
                </th>
                <th
                  class="css-5r049v"
                >
                  <label
                    for="dueDate"
                  >
                    dueDate
                  </label>
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td
                  class="css-l9u794"
                >
                  <input
                    class="css-1lxapgb"
                    id="title"
                    name="title"
                    placeholder="買い物に行く"
                    type="text"
                  />
                </td>
                <td
                  class="css-l9u794"
                >
                  <input
                    class="css-1lxapgb"
                    id="description"
                    name="description"
                    placeholder="牛乳、卵を購入する"
                    type="text"
                  />
                </td>
                <td
                  class="css-l9u794"
                >
                  <input
                    class="css-1lxapgb"
                    id="dueDate"
                    name="dueDate"
                    type="date"
                  />
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="css-8c9rmk"
          >
            <button
              class="css-e26wfq"
              type="submit"
            >
              タスクを追加
            </button>
          </div>
        </form>
      </section>
      <section>
        <h2
          class="css-aqmdkm"
        >
          一覧
        </h2>
        <table
          class="css-tv8nwf"
        >
          <thead
            class="css-14fr7p1"
          >
            <tr>
              <th
                class="css-5r049v"
              >
                id
              </th>
              <th
                class="css-5r049v"
              >
                title
              </th>
              <th
                class="css-5r049v"
              >
                description
              </th>
              <th
                class="css-5r049v"
              >
                dueDate
              </th>
              <th
                class="css-5r049v"
              >
                isCompleted
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="css-l9u794"
              >
                1
              </td>
              <td
                class="css-l9u794"
              >
                Task 1
              </td>
              <td
                class="css-l9u794"
              >
                Description 1
              </td>
              <td
                class="css-l9u794"
              >
                2022-01-01
              </td>
              <td
                class="css-l9u794"
              >
                false
              </td>
            </tr>
            <tr>
              <td
                class="css-l9u794"
              >
                2
              </td>
              <td
                class="css-l9u794"
              >
                Task 2
              </td>
              <td
                class="css-l9u794"
              >
                Description 2
              </td>
              <td
                class="css-l9u794"
              >
                2022-01-02
              </td>
              <td
                class="css-l9u794"
              >
                true
              </td>
            </tr>
          </tbody>
        </table>
      </section>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
